src/components/course/enrollment/components/ViewOnDashboard.jsx

```diff
@@ -0,0 +1,28 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { Link } from 'react-router-dom';
+
+import { AppContext } from '@edx/frontend-platform/react';
+import { EnrollButtonCta } from '../common';
+
+import { enrollLinkClass } from '../constants';
+
+// view on dashboard
+const ViewOnDashboard = ({ enrollLabel }) => {
```

Comment:
This component is almost identical to the `ToDataSharingConsentPage` component, and very similar to the `ToCourseWarePage` component. I'm not sure why these are all separate components; can you say more about your architecture here?

Reply:
indeed, there is only one input param different from the toDSC component case. I figured it's best to keep them separate for reasons of clarity despite their implementation looking fairly close. also looks like there is another small difference (based on prior code, in how the link is rendered ('a' vs router 'Link')

Reply:
Sorry did not explain well, so I chose to go with clarity of naming and did not bother to handle the last remaining pieces of DRYness since it looked like there were going to be minute differences. I did make one exception when the code for two cases was identical 100%, we can revisit that later if needed.